bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java

```diff
@@ -0,0 +1,167 @@
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.link;
+
+import static com.adobe.cq.wcm.core.components.commons.link.LinkConstants.PN_LINK_TARGET;
+import static com.adobe.cq.wcm.core.components.commons.link.LinkConstants.PN_LINK_URL;
+
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.adobe.cq.wcm.core.components.commons.link.Link;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Simple implementation for resolving and validating links from model's resources.
+ * This is a Sling model that can injected into other models using the <code>@Self</code> annotation.
+ */
+@Model(adaptables=SlingHttpServletRequest.class)
+public class LinkHandler {
+
+    /**
+     * List of allowed/supported values for link target.
+     * <code>_self</code> is used in the edit dialog but not listed as allowed here as we do not
+     * want to render a target attribute at all when <code>_self</code> is selected.
+     */
+    private static final Set<String> VALID_LINK_TARGETS = ImmutableSet.of("_blank", "_parent", "_top");
+
+    
+    @Self
+    private SlingHttpServletRequest request;
+    @ScriptVariable
+    private PageManager pageManager;
+    
+    /**
+     * Resolves a link from the properties of the given resource.
+     * @param resource Resource
+     * @return Link may be invalid, but is never null
+     */
+    public @NotNull Link getLink(@NotNull Resource resource) {
+        return getLink(resource, PN_LINK_URL);
+    }
+
+    /**
+     * Resolves a link from the properties of the given resource.
+     * @param resource Resource
+     * @param linkURLPropertyName Property name to read link URL from.
+     * @return Link may be invalid, but is never null
+     */
+    public @NotNull Link getLink(@NotNull Resource resource, String linkURLPropertyName) {
```

Comment:
isn't `PN_LINK_URL` the only property name allowed to store a link? If so this method is not needed.
What's the use case for having other properties which store a link URL?

Reply:
I just realised that we currently have different property names (link, linkURL) in the core components. Having above method would help keeping backwards compatibility. So let's keep it.